Add proper rbac to run patroni on k8s

soltysh opened the pull request:
This PR adds 3 resources that will properly setup the RBAC:

1. service account, which is also assigned to the pods of the cluster, so that they use those particular permissions 
2. a role, which holds only the necessary permissions that patroni members need to interact with k8s cluster
3. a rolebinding, which connects two two former things together to use.

The role and rolebinding was created using this tool https://github.com/liggitt/audit2rbac which looks at [audit logs](https://kubernetes.io/docs/tasks/debug-application-cluster/audit/#advanced-audit) provided  by the api server.

soltysh commented:
@jberkus @k1-hedayati ptal

CyberDem0n commented on kubernetes/patroni_k8s.yaml:
```diff
@@ -120,3 +121,60 @@ type: Opaque
+  - patronidemo
+  resources:
+  - endpoints
+  verbs:
```
Patroni also needs to be able to `list` and `watch` Endpoints if `kubernetes.use_endpoint: true`
Also in this case it doesn't need have an access to `ConfigMaps`.

soltysh replied:
Hmmm... The options are:
1. document both approaches with roles specific for each use case
2. create uber role with all in
wdyt?

soltysh replied:
For the demo example I think I'd go with the latter with comment that this is for configmap approach and the other is for endpoints.

jberkus replied:
why not do the grants for both things in the default roles/template, and document how to restrict them based on which mode you're running in?

soltysh replied:
Yeah, I'll update it later today. That's I guess the best approach.

CyberDem0n commented on kubernetes/patroni_k8s.yaml:
```diff
@@ -120,3 +121,60 @@ type: Opaque
+  - list
+  - patch
+  - update
+  - watch
```
Patroni never watch on `Pods`

soltysh replied:
It's more of an approach we in OpenShift take, that whenever you have list a watch is just there. That's also what's represented in the tool I've mentioned and why this is here.

soltysh replied:
From a security pov having either or both is almost the same.

soltysh commented:
I've updated the roles to include the ones needed for endpoints, and added appropriate comment wrt when they are needed. I've also dropped one more limitation, previously there was an exact name for an endpoint, which will only confuse users.

soltysh commented:
@CyberDem0n ptal

alexeyklyukin commented:
👍

CyberDem0n commented:
👍

CyberDem0n commented:
Merged, thank you @soltysh
